stage/Game.tsx

```diff
@@ -10,22 +10,42 @@ const Game = () => {
   const [opponentBoard, setOpponentBoard] = useState<number[][]>([]);
   const [playerBoard, setPlayerBoard] = useState<number[][]>([]);
 
+  const Ships: Ship[] = [
+    {
+      type: ShipNames["Destroyer"],
+    },
+    {
+      type: ShipNames["Submarine"],
+    },
+    {
+      type: ShipNames["Cruiser"],
+    },
+    {
+      type: ShipNames["Battleship"],
+    },
+    {
+      type: ShipNames["Carrier"],
+    },
+  ];
+
   const width = 10;
 
-  function createBoard() {
+  // make empty board for both players
+  function createBoard(): number[][] {
```

Comment:
While the function does create a board, the actual function of the function - as in, what role this function serves, is returning a board.  So its name should be `newBoard`.

Which in that case, it should just be a constant.

Reply:
This function also only runs under `useEffect`. It should exist inside it.

Reply:
What could be the reason for the function existing inside the `useEffect`?

Reply:
Anything outside of a hook regenerates on every re-render. In this case, this function would be recreated on every render. That is unnecessary, because the function has the exact same function every time. We'd rather have it the same way.

Putting it in `useEffect` not only prevents that, but in this particular `useEffect` (this one only runs once) removes the function from memory after the first run.

It also makes sure collaborators don't try to use it outside of its actual use case. This function, for now, only exists to do what `useEffect` needs to do, so it doesn't need to sit outside of it as if it's used anywhere else.

Reply:
Well noted. That makes sense